core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -650,8 +672,9 @@ export class ExplorationEditor extends BaseUser {
     });
     await Promise.all([
       this.clickOn(confirmDiscardButton),
-      this.page.waitForNavigation(),
+      this.page.waitForNavigation({waitUntil: 'networkidle0'}),
```

Comment:
Would it be better to have waitForNavigation() always wait until the network is idle, by default? (Not sure -- what do you think?)

Reply:
I think they both have their specific uses -
Here are some scenarios where waitForNavigation() might be preferred over {waitUntil: 'networkidle0'}:

Quick Navigation: If you want to wait for navigation to complete without waiting for all network activity to stop, waitForNavigation() without specifying {waitUntil: 'networkidle0'} is more appropriate. This is useful for scenarios where you don't need to ensure that all resources are fully loaded before proceeding.

Non-Standard Navigation: There may be cases where navigation doesn't involve standard network requests/responses, such as navigating within a single-page application (SPA) where content is loaded dynamically. In such cases, waiting for network idleness might not accurately reflect when navigation is complete, and waitForNavigation() provides a more general solution.

Specific Timing Requirements: If you have specific timing requirements for waiting for navigation, such as waiting until DOMContentLoaded or load events, you might prefer using waitForNavigation() with the appropriate waitUntil option (e.g., 'domcontentloaded', 'load') rather than relying solely on network idleness.

By defaulting to {waitUntil: 'networkidle0'}, you're essentially making your navigation waits more conservative, ensuring that the page is fully loaded and has minimal network activity before proceeding. While this can be suitable for many scenarios, there may still be cases where you prefer the flexibility or specificity offered by waitForNavigation() without specifying network idleness.

Reply:
OK, thanks -- I guess we're moving towards an SPA so let's leave it as you have it for now.